src/derivative.jl

```diff
@@ -37,6 +37,22 @@ end
     return out
 end
 
+@inline function derivative!(::ImmutableDiffResult, f::F, x::R) where {F,R<:Real}
+    T = typeof(Tag(F, R))
+    ydual = f(Dual{T}(x, one(x)))
+    out = DiffBase.DiffResult(value(ydual), ForwardDiff.partials(ydual, 1))
+    return out
+end
+
+@inline function derivative!(::ImmutableDiffResult, f!::F, y, x::R, cfg::AllowedDerivativeConfig{F,H} = DerivativeConfig(f!, y, x)) where {F,R<:Real,H}
+    ydual = cfg.duals
+    seed!(ydual, y)
+    f!(ydual, Dual{Tag{F,H}}(x, one(x)))
+    map!(value, y, ydual)
+    out = DiffResult(map(value, ydual), extract_derivative(ydual))
+    return out
+end
```

Comment:
Instead of adding new methods here, I think you can just update `extract_value!` and `extract_derivative!` callsites to alias the output, right?